### PR TITLE
Added: Rating slider

### PR DESCRIPTION
**Pull request by saurabhchirde**

### CODE REVIEW

File: src/Utils/FilterFunctions/filterByRating.js [hunk 1]
```diff
@@ -23,8 +24,10 @@ const filterByRating = (filteredArray, state) => {
   if (state.byRating.fourStar) {
     tempArray = withoutFiltered.filter((prod) => prod.rating >= 4);
   }
+  if (state.byRating.fiveStar) {
+    tempArray = withoutFiltered.filter((prod) => prod.rating >= 4.8);
+  }
```
Comment (manojsarna) [inline src/Utils/FilterFunctions/filterByRating.js:29]: Suggestion : could have done this without using a temp Array , return the destructured withoutFiltered Array

File: src/Components/ProductListingPage/FilterSection/RatingSection/RatingSection.jsx [hunk 2]
```diff
@@ -1,54 +1,49 @@
...
+        <span className="rating-badge-star">
+          <i className={byRating.threeStar ? "fas fa-star" : "far fa-star"}></i>
+        </span>
+        <span className="rating-badge-star">
+          <i className={byRating.fourStar ? "fas fa-star" : "far fa-star"}></i>
+        </span>
+        <span className="rating-badge-star">
+          <i className={byRating.fiveStar ? "fas fa-star" : "far fa-star"}></i>
+        </span>
+      </div>
```
Comment (manojsarna) [inline src/Components/ProductListingPage/FilterSection/RatingSection/RatingSection.jsx:34]: Appreciation : Rating Section is looking good :)

**Review by manojsarna — APPROVED** [review]
UI Review : Rating Section is looking good

Overall code is good

Do Check the comments !!

File: src/Components/ProductListingPage/FilterSection/RatingSection/RatingSection.jsx (hunk 2 above)
Comment (PuneetP16) [inline src/Components/ProductListingPage/FilterSection/RatingSection/RatingSection.jsx:34]: Appreciation: Now the Rating component has become much more pleasing to both the user and developer (less code and easy to manage).

Suggestion: You could make a small component for each star here as most of the values looks repetitive. Reducing 10 lines of code to 2-3 lines

**Review by PuneetP16 — APPROVED** [review]
UI has improved, check the suggestion, looks good to merge.

File: src/Utils/FilterFunctions/filterByRating.js (hunk 1 above)
Comment (saurabhchirde) [reply src/Utils/FilterFunctions/filterByRating.js:29]: Ok, Noted

**Pull request merged**